platform/source/mbed_retarget.cpp

```diff
@@ -64,7 +64,11 @@ asm(" .global __use_full_stdio\n");
 #elif defined(__ICCARM__)
 #   include <yfuns.h>
 #   define PREFIX(x)        _##x
-#   define OPEN_MAX         16
+#   if !MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY
```

Comment:
You've done this only for IAR.

For GCC `OPEN_MAX` actually comes from header files we don't control, which is why I was suggesting a separate local define.